Doctrine/Orm/Filter/SearchFilter.php

```diff
@@ -152,4 +162,33 @@ private function getFilterValueFromUrl($value)
 
         return $value;
     }
+
+    /**
+     * Gets query from strategy.
+     *
+     * @param  string $strategy
+     * @param  string $property
+     * @param  string $value 
+     *
+     * @return string
+     *
+     * @throws \DomainException If strategy not found
+     */
+    private function getQueryFromStrategy($strategy, $property, $value) {
+        switch ($strategy) {
+            case self::STRATEGY_EXACT:
+                return sprintf('o.%s = \'%s\'', $property, $value);
+            
+            case self::STRATEGY_PARTIAL:
+                return sprintf('o.%s LIKE \'%%%s%%\'', $property, $value);
+
+            case self::STRATEGY_STARTS_WITH:
+                return sprintf('o.%s LIKE \'%s%%\'', $property, $value);
+
+            case self::STRATEGY_ENDS_WITH:
+                return sprintf('o.%s LIKE \'%%%s\'', $property, $value);
+        }
+
+        throw new \DomainException(sprintf('strategy %s does not exist.', $strategy));
```

Comment:
Following @dunglas's comment, I'll add that a domain exception means an exception that is part of the library :)